src/sentry/static/sentry/app/utils/withSentryAppComponents.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import getDisplayName from 'app/utils/getDisplayName';
+import SentryAppComponentsStore from 'app/stores/sentryAppComponentsStore';
+
+const withSentryAppComponents = type => WrappedComponent =>
```

Comment:
Can we make type an option instead of having it return another function?

So the interface could be like 

```py
withSentryAppComponents(Frame, {componentType: 'stacktrace-link'})
```

Then the function here would be 

```py
const withSentryAppComponents = (WrappedComponent, {componentType}) =>
```

You could even have it return all components then if you don't pass `componentType`.

Reply:
@EvanPurkhiser Either approach is fine with me but I was going off of @billyvg's suggestion. Would like his input on this.

Reply:
👍 	- prefer the object as an argument because it gives the reader some context about what it is

Reply:
@billyvg @EvanPurkhiser I have updated the PR with the second argument as an object